src/fn/upload-docs.js

```diff
@@ -93,20 +97,98 @@ const execute = async () => {
     }
   };
 
-  return processNextBatch(filesToUpload, INITIAL_BATCH_SIZE);
-};
+  return processNextBatch(filenamesToUpload, INITIAL_BATCH_SIZE);
+}
 
-const getErrorsWhereDocIdDiffersFromFilename = filePaths =>
-  filePaths
+function analyseFiles(filePaths) {
+  return filePaths
     .map(filePath => {
       const json = fs.readJson(filePath);
       const idFromFilename = path.basename(filePath, FILE_EXTENSION);
 
       if (json._id !== idFromFilename) {
-        return `File '${filePath}' sets _id:'${json._id}' but the file's expected _id is '${idFromFilename}'.`;
+        return { error: `File '${filePath}' sets _id:'${json._id}' but the file's expected _id is '${idFromFilename}'.` };
+      }
+
+      if (json._deleted && json.cht_disable_linked_users) {
+        return { delete: json._id };
       }
     })
-    .filter(err => err);
+    .filter(Boolean);
+}
+
+async function handleUsersAtDeletedFacilities(deletedDocIds) {
+  await assertCoreVersion();
+  
+  const affectedUsers = await getAffectedUsers(deletedDocIds);
+  const usernames = affectedUsers.map(userDoc => userDoc.username).join(', ');
+  if (affectedUsers.length === 0) {
+    trace('No users found needing an update.');
+    return;
+  }
+
+  userPrompt.warnPromptAbort(`This operation will update ${affectedUsers.length} user accounts: ${usernames} and cannot be undone. Are you sure you want to continue?`);
+  await updateAffectedUsers(affectedUsers);
+}
+
+async function assertCoreVersion() {
+  const actualCoreVersion = await getValidApiVersion();
+  if (semver.lt(actualCoreVersion, '4.7.0-dev')) {
+    throw Error(`CHT Core Version 4.7.0 or newer is required to use --disable-users options. Version is ${actualCoreVersion}.`);
```

Comment:
question: I've noticed an inconsistency in the command structure. The `disable-users` argument was moved from the `disable-users` command to the `merge-contacts` command. However, after reviewing the `merge-contacts` command, I don't see a direct invocation of `upload-docs`. 

Are we looking at a scenario where these commands should be combined, such as `cht merge-contacts upload-docs --disable-users`? 

My concern is that running this against a CHT 4.6.0 instance might result in an error, even if the `--disable-users` flag isn't explicitly passed.

Reply:
> My concern is that running this against a CHT 4.6.0 instance might result in an error, even if the --disable-users flag isn't explicitly passed.

You're right. Sorry that was a bit sloppy. I guess we want the `--disable-users` gate on both actions.

Reply:
Yes, option 1 is the best option. Let's go with that.

Reply:
OK. That is what is in this PR now.